featuretools/computational_backends/feature_set_calculator.py

```diff
@@ -936,7 +939,16 @@ def update_feature_columns(feature_data, data):
         data.update(new_cols)
         return data
 
-    return data.assign(**new_cols)
+    if isinstance(data, pd.DataFrame):
+        return pd.concat([data, pd.DataFrame(new_cols, index=data.index)], axis=1)
+    else:
+        for name, col in new_cols.items():
+            col.name = name
+            if isinstance(data, dd.DataFrame):
+                data = dd.concat([data, col], axis=1)
+            else:
```

Comment:
Should we have an explicit check to see if `ps` is defined and if `isinstance(data, ps.DataFrame)`?

The case I am thinking about is, if data is a `pd.Series` for example, it would fall into this else block.